File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        int ct = 0;
        for (int[] cmd:commands) {
            int i = cmd[0];
            int j = cmd[1];
            int k = cmd[2];
            int[] temp = new int[j-i+1];
            
            for (int idx=i-1;idx<j;idx++) {
                temp[idx-(i-1)] = array[idx];
            }
            Arrays.sort(temp);
            answer[ct++] = temp[k-1];
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for (vector<int> cmd : commands) {        
        int start = cmd[0] - 1;
        int end = cmd[1]; // 포함하지 않는 끝 위치
        int k = cmd[2] - 1;
        
        vector<int> sliced(array.begin() + start, array.begin() + end);
        sort(sliced.begin(), sliced.end());
        
        answer.push_back(sliced[k]);
    }
    
    return answer;
}